core/src/action.rs

```diff
@@ -237,8 +237,11 @@ pub struct GetLinksKey {
     /// The address of the Link base
     pub base_address: Address,
 
+    /// The link type
+    pub link_type: String,
+
     /// The link tag
-    pub tag: String,
+    pub tag: String, // this makes it tricky to retrieve all links
```

Comment:
I believe this is causing it to timeout while waiting for a get_links response to return when the returning tag is different (e.g. like when the wildcard tag is used)